src/tequila/circuit/compiler.py

```diff
@@ -1297,3 +1315,83 @@ def compile_trotterized_gate(gate, compile_exponential_pauli: bool = False):
         return compile_exponential_pauli_gate(result)
     else:
         return result
+
+
+@compiler
+def compile_ry(gate: RotationGateImpl, controlled_rotation: bool = False) -> QCircuit:
+    """
+    Compile Ry gates into Rx and Rz.
+    Parameters
+    ----------
+    gate:
+        the gate.
+    controlled_rotation:
+        to determine if the gate break down need to be applied here
+
+    Returns
+    -------
+    QCircuit, the result of compilation.
+    """
+    if gate.name.lower() == "ry":
+
+        if not (gate.is_controlled() and controlled_rotation):
+
+            return Rz(target=gate.target, control=None, angle=-numpy.pi / 2) \
+                   + Rx(target=gate.target, control=gate.control, angle=gate.parameter) \
+                   + Rz(target=gate.target, control=None, angle=numpy.pi / 2)
+
+    return QCircuit.wrap_gate(gate)
+
+
+@compiler
+def compile_y(gate: QGateImpl) -> QCircuit:
+    """
+    Compile Y gates into X and Z.
+    Parameters
+    ----------
+    gate:
+        the gate.
+
+    Returns
+    -------
+    QCircuit, the result of compilation.
+    """
+    if gate.name.lower() == "y":
+
+        return Rz(target=gate.target, control=None, angle=-numpy.pi / 2) \
+               + Rx(target=gate.target, control=gate.control, angle=numpy.pi) \
+               + Rz(target=gate.target, control=None, angle=numpy.pi / 2)
+
+    else:
+        return QCircuit.wrap_gate(gate)
+
+
+@compiler
+def compile_ch(gate: QGateImpl) -> QCircuit:
+    """
+    Compile H gates into its equivalent:
+        ch a,b = h b + sdg b + cx a,b + h b + t b + cx a,b + t b + h b + s b + x b + s a
+    Parameters
+    ----------
+    gate:
+        the gate.
+
+    Returns
+    -------
+    QCircuit, the result of compilation.
+    """
+    if gate.name.lower() == "h" and gate.is_controlled():
+
```

Comment:
need to assert that the hadamard gate is not parametrized
they can have powers which is then the H^power gate (e.g. tq.gates.H(target=0, power=1.5))

of the power needs to be carried over into the decomposition (can't see it directly from here)

Reply:
similar as "angle" for Ry

Reply:
I think the H decomposition can be written as
H^a = Ry(0.25*pi) Z^a Ry(-0.25*pi)
The controlled version is then the same with a controlled-Z in the middle 
but better check if that is correct

Logic behind is:
Ry(0.25*pi) transforms from the 0,1 basis into the eigenbasis of H
In the eigenbasis of H it acts as Z

Reply:
You are correct: `H^a = Ry(0.25pi) Z^a Ry(-0.25pi)`
[Here](https://www.wolframalpha.com/input/?i=%7B%7Bcos%28pi%2F8%29%2C-sin%28pi%2F8%29%7D%2C%7Bsin%28pi%2F8%29%2Ccos%28pi%2F8%29%7D%7D+*+%7B%7B1%2C0%7D%2C%7B0%2C-1%7D%7D+*+%7B%7Bcos%28pi%2F8%29%2Csin%28pi%2F8%29%7D%2C%7B-sin%28pi%2F8%29%2Ccos%28pi%2F8%29%7D%7D) the matrix equivalence

I changed this equivalence, so the power argument is considered.